ambari-common/src/main/python/ambari_commons/shell.py

```diff
@@ -104,10 +104,11 @@ class RepoCallContext(object):
   use_repos = None
   skip_repos = None
   is_upgrade = False
+  action_force = False  # currently only for install action
 
   def __init__(self, ignore_errors=True, retry_count=2, retry_sleep=30, retry_on_repo_unavailability=False,
                retry_on_locked=True, log_output=True, use_repos=None, skip_repos=None,
-               is_upgrade=False):
+               is_upgrade=False, action_force=False):
```

Comment:
Unlike `is_upgrade`, I don't see `action_force` being set anywhere.  Can you please clarify if that's intentional?

Reply:
`is_upgrade` have used before, when we have installed packages using asterisk and packages had same names in different stacks. So when we have moved to newer stack, installation supposed to be forces, even if such package have existed. For 2-3 years already, we using conventional package naming, which is different for each build and no need to force re-installation of packages if they have already installed. 

 However, i still left possibility for 3d-part stack developers to use previous behavior if it intended by their conventions.

Reply:
Thanks!